From 2bb182f041b898813499a6604181a82fe3ac735c Mon Sep 17 00:00:00 2001
From: Bert van Hall <bert.vanhall@avionic-design.de>
Date: Tue, 13 Dec 2016 10:26:00 +0100
Subject: [PATCH] avoid lib<arch> as libdir and use standard include path

According to the comment in configure.ac, this should only have been
used when cross-building for WIN32. Since the build-tools does not care
about multiarch libraries, remove the detection and just set toolexecdir
to libdir.

Set the include path to the standard directory instead of some obscure
lnidir subdirectory.

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 configure.ac        | 21 +--------------------
 include/Makefile.am |  3 +--
 libffi.pc.in        |  2 +-
 3 files changed, 3 insertions(+), 23 deletions(-)

diff --git a/configure.ac b/configure.ac
index a7bf5ee..fe6f905 100644
--- a/configure.ac
+++ b/configure.ac
@@ -590,26 +590,7 @@ AC_ARG_ENABLE(purify-safety,
     AC_DEFINE(USING_PURIFY, 1, [Define this if you are using Purify and want to suppress spurious messages.])
   fi)
 
-# These variables are only ever used when we cross-build to X86_WIN32.
-# And we only support this with GCC, so...
-if test "x$GCC" = "xyes"; then
-  if test -n "$with_cross_host" &&
-     test x"$with_cross_host" != x"no"; then
-    toolexecdir="${exec_prefix}"/'$(target_alias)'
-    toolexeclibdir="${toolexecdir}"/lib
-  else
-    toolexecdir="${libdir}"/gcc-lib/'$(target_alias)'
-    toolexeclibdir="${libdir}"
-  fi
-  multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
-  AC_SUBST(toolexecdir)
-else
-  toolexeclibdir="${libdir}"
-fi
+toolexeclibdir="${libdir}"
 AC_SUBST(toolexeclibdir)
 
 AC_CONFIG_COMMANDS(include, [test -d include || mkdir include])
diff --git a/include/Makefile.am b/include/Makefile.am
index fd28024..7a0d02f 100644
--- a/include/Makefile.am
+++ b/include/Makefile.am
@@ -5,5 +5,4 @@ AUTOMAKE_OPTIONS=foreign
 DISTCLEANFILES=ffitarget.h
 EXTRA_DIST=ffi.h.in ffi_common.h
 
-includesdir = $(libdir)/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
-nodist_includes_HEADERS = ffi.h ffitarget.h
+nodist_include_HEADERS = ffi.h ffitarget.h
diff --git a/libffi.pc.in b/libffi.pc.in
index edf6fde..6fad83b 100644
--- a/libffi.pc.in
+++ b/libffi.pc.in
@@ -2,7 +2,7 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 toolexeclibdir=@toolexeclibdir@
-includedir=${libdir}/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includedir=@includedir@
 
 Name: @PACKAGE_NAME@
 Description: Library supporting Foreign Function Interfaces
-- 
2.7.4

